src/peft/utils/loftq_utils.py

```diff
@@ -227,3 +233,139 @@ def loftq_init(weight: Union[torch.Tensor, torch.nn.Parameter], num_bits: int, r
     lora_A, lora_B = R, L
 
     return dequantized_weight.to(device=device, dtype=dtype), lora_A, lora_B
+
+
+def _check_model_path_loftq(model_path, peft_model):
+    """Check if there is a local model file corresponding to the given model, then format it properly."""
+    if model_path is None:
+        try:
+            model_path = snapshot_download(peft_model.base_model.config._name_or_path, local_files_only=True)
+        except AttributeError as exc:
+            raise ValueError(
+                "The provided model does not appear to be a transformers model. In this case, you must pass the "
+                "model_path to the safetensors file."
+            ) from exc
+        except LocalEntryNotFoundError as exc:
+            raise ValueError("The model.safetensors file must be present on disk, but it could not be found.") from exc
+
+    suffix = "model.safetensors"
```

Comment:
should we support `.bin` as well for BC - weights that are pushed on the hub with .bin extension?

Reply:
Unfortunately, I don't think this can work. Or at least it would require a lot of extra effort to make .bin work. The problem is that I want to load one weight at a time to reduce memory overhead, which works with safetensors but not with pickle. For the latter, we would load a whole new copy of the model into memory.

Reply:
oh ok, then sounds good !